src/Discord.Net/Utilities/EmoteUtilities.cs

```diff
@@ -7,21 +7,28 @@ public static class EmoteUtilities
         public static string FormatGuildEmote(ulong id, string name)
             => $"<:{name}:{id}>";
 
-        public static (ulong, string) ParseGuildEmote(string formatted)
+        // TODO: perf: bench whether this should be passed by ref (in)
+        public static bool TryParseGuildEmote(ReadOnlySpan<char> formatted, out (ulong, string) result)
         {
-            if (formatted.IndexOf('<') != 0 || formatted.IndexOf(':') != 1 || formatted.IndexOf('>') != formatted.Length-1)
-                throw new ArgumentException("passed string does not match a guild emote format", nameof(formatted)); // TODO: grammar
+            result = default;
 
-            int closingIndex = formatted.IndexOf(':', 2);
+            if (formatted.IndexOf('<') != 0 || formatted.IndexOf(':') != 1 || formatted.IndexOf('>') != formatted.Length - 1)
```

Comment:
We should use something like this instead here:
```cs
if (formatted.Length < 7)
    return false;
if (formatted[0] != '<' && formatted[1] == ':' && formatted[formatted.Length-1] != '>')
    return false;
```
This avoids a O(n) lookup for each character, since we know their positions anyway. The third colon will still have to use IndexOf, however.

Reply:
The 7 here is *AFAIK* the minimum length of a valid emote: `<:aa:0>` I personally don't know of any emote which is shorter than this.

Reply:
The span indexer throws if you try to index an out-of-range value - the length check ensures the indexers work later.

Reply:
> ```cs
> if (formatted[0] != '<' && formatted[1] == ':' && formatted[formatted.Length-1] != '>')
> ```

Is this logic right, like shouldn't there be `||`'s in there instead?

Reply:
Err, Probably. That `==` should likely also be a `!=` too.